eo-runtime/src/main/java/org/eolang/Attr.java

```
@@ -68,4 +68,12 @@ public interface Attr {
      * @param phi The object to put
      */
     void put(Phi phi);
+
+    /**
+     * Put a new object with position of the attribute in.
+     *
+     * @param pos The position of the attribute
+     * @param phi The object to put
+     */
+    void put(int pos, Phi phi);
```

@Suban05 I don't think we need this method in the interface, we are able to decide either we can put attribute or not in `PhDefault`

---

@maxonfjvipon it seems that I didn't understand the task correctly and implemented it more difficult than necessary. Do I understand correctly that we should throw this exception with a new message when pos > 0?

---

@Suban05 no, I suggest you to do something like this:
```java
class PhDefault ... {
    @Override
    public void put(final int pos, final Phi object) {
        final String name = this.attr(pos);
        if (!this.attrs.containsKey(name)) {
            throw new ExUnset(
                String.format(
                    "Can't #put(\"%s\", %s) to %s, because the attribute is absent",
                    name, object, this
                )
            );
        }
        if (!this.attrs.get(name) instanceof AtVoid ) {
            throw new ExReadOnly(
               "Can't set non void attribute blah blah blah"
            )
        }
        this.put(this.attr(pos), object);
    }
}
```

I think such simple `instanceof` is enough, no need to add this overhead with extra method in `Attr` interface. Maybe you'll require one more private method in `PhDefault` to reduce code duplication where `ExUnset` is thrown, but it's up to you

---

@maxonfjvipon I got you, thanks a lot

I added a new method to the interface because there is a task to remove the Attr interface https://github.com/objectionary/eo/issues/3480, which needs to be replaced with Phi. this interface requires the implementation of the put method with position, so I thought this approach would make the move to the new interface more lenient